tools/wptrunner/wptrunner/wptmanifest/parser.py

```diff
@@ -558,31 +581,54 @@ def expect(self, type, value=None):
 
         self.consume()
 
+    def maybe_consume_inline_comment(self):
+        if self.token[0] == token_types.inline_comment:
+            self.comments.append(self.token)
+            self.consume()
+
+    def consume_comments(self):
+        while self.token[0] == token_types.comment:
+            self.comments.append(self.token)
+            self.consume()
+
+    def flush_comments(self, node=None):
```

Comment:
Done. Attaching comments to a child list/string node is actually not the reason for comment buffering. Comment buffering is for handling when the target `DataNode`/`KeyValueNode` has not been created yet because their tokens appear later in the stream.